### DISCUSSION TITLE
How to show toc for a markdown content using mdx?

### DISCUSSION THREAD

**Discussion by shubdev07** [mdx-js/mdx#1410] [Q&A] [answered]
I was wondering how do i show a toc for a markdown file content using this library?

**Comment by ChristianMurphy** [comment, marked as answer]
Using a remark plugin:
https://mdxjs.com/advanced/transform-content#transform-content
https://github.com/remarkjs/remark-toc

**Reply by helloworld** [reply]
Do you have any suggestions/tips on how to extract the table of contents instead of transforming the existing content to have a table of contents?

**Reply by ChristianMurphy** [reply]
https://github.com/remarkjs/remark/discussions/568#discussion-32545

**Comment by helloworld** [comment]
For anyone else who lands on this issue and needs to extract the table of contents from mdx so that they can display it elsewhere on their page (as opposed to transforming and including the markdown inline using `remark-toc`), this is how I was able to do it: 

https://github.com/dagster-io/dagster/blob/0110-docs/docs/next/pages/%5B...page%5D.tsx#L338-L341
